assets/src/stories-editor/plugins/story-settings-panel.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { metaToAttributeNames } from '../helpers';
+
+/**
+ * WordPress dependencies
+ */
+import { PluginDocumentSettingPanel } from '@wordpress/edit-post';
+import { __ } from '@wordpress/i18n';
+import { compose } from '@wordpress/compose';
+import {
+	withSelect,
+	withDispatch,
+} from '@wordpress/data';
+import {
+	SelectControl,
+	RangeControl,
+} from '@wordpress/components';
+
+const MetaFields = ( props ) => {
+	const {
+		meta,
+		autoAdvanceAfterOptions,
+		updateMeta,
+	} = props;
+
+	const {
+		autoAdvanceAfter,
+		autoAdvanceAfterDuration,
+	} = metaToAttributeNames( meta );
+
+	const currentOption = autoAdvanceAfterOptions.find( ( i ) => i.value === autoAdvanceAfter ) || {};
+
+	return (
+		<>
+			<p>{ __( 'These settings are applied to new pages.', 'amp' ) }</p>
+			<SelectControl
+				label={ __( 'Advance to next page', 'amp' ) }
+				help={ currentOption.description || '' }
+				value={ autoAdvanceAfter }
+				options={ autoAdvanceAfterOptions }
+				onChange={ ( value ) => updateMeta( { amp_story_auto_advance_after: value } ) }
+			/>
+			{ 'time' === autoAdvanceAfter && (
+				<RangeControl
+					label={ __( 'Time in seconds', 'amp' ) }
+					value={ autoAdvanceAfterDuration ? parseInt( autoAdvanceAfterDuration ) : 0 }
+					onChange={ ( value ) => updateMeta( { amp_story_auto_advance_after_duration: value } ) }
+				/>
+			) }
+		</>
+	);
+};
+
+MetaFields.propTypes = {
+	autoAdvanceAfterOptions: PropTypes.array.isRequired,
+	updateMeta: PropTypes.func.isRequired,
+	meta: PropTypes.shape( {
+		amp_story_auto_advance_after: PropTypes.string,
+		amp_story_auto_advance_after_duration: PropTypes.number,
+	} ),
+};
+
+const EnhancedMetaFields = compose(
+	withSelect( ( select ) => {
+		const { getEditedPostAttribute } = select( 'core/editor' );
+		const { getSettings } = select( 'amp/story' );
+		const { storySettings } = getSettings();
+		const { autoAdvanceAfterOptions } = storySettings || {};
+
+		return {
+			meta: getEditedPostAttribute( 'meta' ),
+			autoAdvanceAfterOptions,
+		};
+	} ),
+	withDispatch( ( dispatch, { meta } ) => {
+		const { editPost } = dispatch( 'core/editor' );
+		return {
+			updateMeta( newMeta ) {
+				editPost( {
+					meta: {
+						...meta,
+						...newMeta,
+					},
+				} );
+			},
+		};
+	} ),
+)( MetaFields );
+
+export const name = 'amp-story-settings-panel';
+
+export const render = () => {
+	return (
+		<PluginDocumentSettingPanel
+			name={ name }
+			title={ __( 'Story Settings', 'amp' ) }
```

Comment:
Is it expected that this adds an icon next to the title? Looks a bit odd:

![Screenshot 2019-09-11 at 19 18 47](https://user-images.githubusercontent.com/841956/64719702-21acd280-d4c9-11e9-8d26-9a57d029be9a.png)

Reply:
@swissspidy Agreed, but it is both expected and somewhat required.

- `registerPlugin` adds this [default icon](https://github.com/WordPress/gutenberg/blob/9fef2e82720b2ddf4d7d8c8bd3a42a24ae721840/packages/plugins/src/api/index.js#L131) to its context.
- `PluginDocumentSettingsPanel` [picks it up](https://github.com/WordPress/gutenberg/blob/aff692edeb78499df84f897f77befb240c5f88e8/packages/edit-post/src/components/sidebar/plugin-document-setting-panel/index.js#L97) as a fallback.

Seems like the simplest way to get rid of the icon there is to provide an incorrect icon name or a blank SVG icon. Neither seems like a great solution. What do you propose?